[integ-tests] improve build-image integration tests

hanwen-cluster opened the pull request:
### Description of changes
* The first 3 commits are cherry-pick from `release-3.12` branch.
* The last 3 commits are to make sure kernel consistency on RHEL/Rocky build-image

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

hanwen-cluster committed:
[Test] Install Nvidia driver in createami tests

Reason: This increases test coverage of dependencies files in S3 buckets

How
1. Use GPU instances and use availability zones with best GPU capacity
2. Change build-image config file

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
[Test] Run build image in China and GovCloud only with alinux2023

This reverts changes from https://github.com/aws/aws-parallelcluster/pull/6450/.
Because first stage AMIs are not available in China and GovCloud. Some OSes require first stage AMIs

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
[Test] Skip Nvidia driver installation if Deep Learning AMI is used

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
[Test] Use latest RHEL/Rocky AMIs

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
[Test] Use pagination to list images

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
[Test] Use UpdateOsPackages when building from vanilla RHEL AMIs

Signed-off-by: Hanwen <hanwenli@amazon.com>

dreambeyondorange commented on tests/integration-tests/tests/common/utils.py:
```diff
@@ -52,8 +52,8 @@
     "rocky8.9": {"name": "Rocky-8-EC2-Base-8.9*", "owners": ["792107900819"]},  # TODO add china and govcloud accounts
-    "rhel9": {"name": "RHEL-9.3*_HVM-*", "owners": RHEL_OWNERS},
-    "rocky9": {"name": "Rocky-9-EC2-Base-9.3*", "owners": ["792107900819"]},  # TODO add china and govcloud accounts
+    "rhel9": {"name": "RHEL-9.*_HVM*", "owners": RHEL_OWNERS},
```
should we be pinning to 9.4 until fsx lustre supports 9.5?

hanwen-cluster replied:
This test has Lustre installation disabled for RHEL/Rocky. So we can go head

dreambeyondorange replied:
I'm not sure if this is what we want.  We aren't going to release for an image that doesn't support lustre, so whats the point of testing it?  I think we should enable lustre and stick with supported OSes

dreambeyondorange commented on tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml:
```diff
@@ -14,17 +14,14 @@ Build:
-            Enabled: false
-{% endif %}
+            # Disable Lustre installation because these newer operating systems release new kernels more often. Lustre usually does not support the latest kernels
+            Enabled: {% if os in ["ubuntu2204", "rhel9", "rocky9"] %} false {% else %} true {% endif %}
```
i wonder if we even care about 9.5 until this is supported.

hanwen-cluster replied:
I agree. Anyway it is good to make the tests more robust

dreambeyondorange replied:
lustre should just always be enabled i think

hanwen-cluster replied:
Done, added another commit

hanwen-cluster committed:
[Test] Only disable Lustre installation if non-first-stage RHEL9/Rocky9/Ubuntu22 AMIs are used

first stage AMIs are used in release.yaml
non-first-stage AMIs are used in develop.yaml

Signed-off-by: Hanwen <hanwenli@amazon.com>
